src/fswAlgorithms/imageProcessing/opticalFlow/opticalFlow.cpp

```diff
@@ -102,40 +102,49 @@ void OpticalFlow::UpdateState(uint64_t CurrentSimNanos)
                                  this->flowMaxLevel,
                                  criteria);
 
-        featurePayload.timeTag_secondImage = this->sensorTimeTag;
-        featurePayload.timeTag_firstImage = this->firstTimeTag;
-
-        v3Copy(this->firstSpacecraftAttitude, featurePayload.sigma_BN_firstImage);
-        v3Copy(navAttBuffer.sigma_BN, featurePayload.sigma_BN_secondImage);
-        v3Copy(this->firstTargetEphemAttitude, featurePayload.sigma_TN_firstImage);
-        v3Copy(ephemMsgBuffer.sigma_BN, featurePayload.sigma_TN_secondImage);
-
-        for (int i=0; i<this->secondFeatures.size(); i++){
-            featurePayload.keyPoints_firstImage[2*i] = this->firstFeatures[i][0];
-            featurePayload.keyPoints_firstImage[2*i+1] = this->firstFeatures[i][1];
-            featurePayload.keyPoints_secondImage[2*i] = this->secondFeatures[i][0];
-            featurePayload.keyPoints_secondImage[2*i+1] = this->secondFeatures[i][1];
-        }
-        featurePayload.cameraID = imageBuffer.cameraID;
-        featurePayload.keyPointsFound = (uint64_t)this->secondFeatures.size();
-        featurePayload.valid = true;
-
-        this->keyPointsMsg.write(&featurePayload, this->moduleID, CurrentSimNanos);
+        /*! If optical flow matches features between the images */
+        if (!this->secondFeatures.empty()){
+            featurePayload.timeTag_secondImage = this->sensorTimeTag;
+            featurePayload.timeTag_firstImage = this->firstTimeTag;
+
+            v3Copy(this->firstSpacecraftAttitude, featurePayload.sigma_BN_firstImage);
+            v3Copy(navAttBuffer.sigma_BN, featurePayload.sigma_BN_secondImage);
+            v3Copy(this->firstTargetEphemAttitude, featurePayload.sigma_TN_firstImage);
+            v3Copy(ephemMsgBuffer.sigma_BN, featurePayload.sigma_TN_secondImage);
+
+            for (int i=0; i<this->secondFeatures.size(); i++){
```

Comment:
:large_orange_diamond: Should this be a `size_t` rather than `int`?

Reply:
Done